docs/source/concepts/settlement.rst

```diff
@@ -7,38 +7,147 @@ Settlement
 :ref:`Settlement <settlement>` refers to the execution of holding transfers originating from
 a financial transaction.
 
-For instance, an example FX spot transaction involves the transfer of a
-EUR-denominated holding from Alice to Bob in exchange for a
-USD-denominated holding.
+The library provides facilities to execute these transfers atomically (i.e., within the same Daml transaction).
+Interfaces are defined in the ``Daml.Finance.Interface.Settlement`` package, whereas implementations are is provided in ``Daml.Finance.Settlement``.
 
-The library provides facilities to execute these transfers atomically
-(i.e., within the same Daml transaction) in the package ``Daml.Finance.Interface.Settlement``.
-An implementation is provided in ``Daml.Finance.Settlement``.
+In this page we illustrate the settlement workflow with the help
+of an example FX transaction, where Alice transfers a EUR-denominated holding
+to Bob, in exchange for a USD-denominated holding of the same amount.
 
-Step
-****
+We then delve into the details of each of the settlement components.
 
-The FX example transaction above contains two steps:
+Workflow
+********
 
-#. transfer EUR from Alice to Bob
-#. transfer USD from Bob to Alice
+Our initial state looks as follows:
 
-They are represented using one ``Step`` each.
-The step defines who is the sender, who is the receiver and what should be transferred (instrument and amount).
+* Alice owns a holding on an `EUR` instrument, for an amount of `1000`
+* Bob owns a holding on a `USD` instrument, for an amount of `1000`
+
+These holding are generally held at different custodians.
+
+.. image:: ../images/settlement_initial_state.png
+   :alt: Alice owns a EUR holding, Bob owns a USD holding.
+
+Instruct
+========
+
+Alice and Bob want to exchange their holdings and agree to enter into the transaction by being signatories on a transaction contract.
+Settlement can then be instructed: this generates
+
+#. an ``Instruction`` to transfer 1000 EUR from Alice to Bob
+#. an ``Instruction`` to transfer 1000 USD from Bob to Alice
+#. a ``Batch`` contract which is used to execute the above Instructions.
+
+.. image:: ../images/settlement_instructed.png
+   :alt: Settlement is instructed.
+
+Each instruction defines, for a step of the settlement chain, who is the sender, who is the receiver and what should be transferred (instrument and amount).
+
+Allocate and Approve
+====================
+
+In order to perform the transfer, we now need to specify for each ``Instruction``
+
+#. which holding should be used
+#. to which account it should be transferred.
+
+Alice and Bob ``allocate`` the instruction where they are, respectively, the sender by pledging their holding.
```

Comment:
Maybe more clear if you describe it only for Alice (and in the next sentence say something like "Bob does the same").